lib/hexpm_web/controllers/package_controller.ex

```diff
@@ -9,6 +9,10 @@ defmodule HexpmWeb.PackageController do
   def index(conn, params) do
     letter = Hexpm.Utils.parse_search(params["letter"])
     search = Hexpm.Utils.parse_search(params["search"])
+    sanitized_search = Hexpm.Search.Sanitizer.sanitize(params["search"])
+    IO.inspect(sanitized_search, label: "sanitized_search")
+    parsed_result = Hexpm.Search.Parser.parse_sanitized_user_input(sanitized_search)
```

Comment:
As for this part the idea I had was to be nice to the user:

1. the sanitizer kicks in, it tries to guess best what the user might have meant, adding additional parentheses where it is clear that they are lacking or removing extra ones.
2. the parser kicks in, it's result is transformed to both: SQL and a new search string, so that the user will see a sanitizer and parsed (aka correct) version, much like a code formatter.